src/integrate/orchestrator.rs

```diff
@@ -0,0 +1,1204 @@
+                std::process::exit(1);
+            }
+        };
+        let _ = event_tx.send(DevnetEvent::ServiceStatus(ServiceStatusData {
+            order: 4,
+            status: Status::Green,
+            name: "bitcoin-explorer".into(),
+            comment: format!("http://0.0.0.0:{}", bitcoin_explorer_port),
+        }));
+
+        match terminator_rx.recv() {
+            Ok(true) => {
+                self.terminate().await;
+            }
+            Ok(false) => {
+                self.restart().await;
+            }
+            _ => {}
+        }
+    }
+
+    // if working_dir empty:
+    //      -> write config files
+    // else
+    //      -> read config files
+
+    pub async fn boot_bitcoin_node_container(&mut self) -> Result<(), String> {
+        let (docker, devnet_config) = match (&self.docker_client, &self.network_config) {
+            (Some(ref docker), Some(ref network_config)) => match network_config.devnet {
+                Some(ref devnet_config) => (docker, devnet_config),
+                _ => return Err("Unable to get devnet configuration".into()),
+            },
+            _ => return Err("Unable to get Docker client".into()),
+        };
+
+        let _info = docker
+            .create_image(
+                Some(CreateImageOptions {
+                    from_image: devnet_config.bitcoin_node_image_url.clone(),
+                    ..Default::default()
+                }),
+                None,
+                None,
+            )
+            .try_collect::<Vec<_>>()
+            .await
+            .map_err(|_| "Unable to create image".to_string())?;
+
+        let mut port_bindings = HashMap::new();
+        port_bindings.insert(
+            format!("{}/tcp", devnet_config.bitcoin_controller_port),
+            Some(vec![PortBinding {
+                host_ip: Some(String::from("0.0.0.0")),
+                host_port: Some(format!("{}/tcp", devnet_config.bitcoin_controller_port)),
+            }]),
+        );
+        port_bindings.insert(
+            format!("{}/tcp", devnet_config.bitcoin_node_p2p_port),
+            Some(vec![PortBinding {
+                host_ip: Some(String::from("0.0.0.0")),
+                host_port: Some(format!("{}/tcp", devnet_config.bitcoin_node_p2p_port)),
+            }]),
+        );
+        port_bindings.insert(
+            format!("{}/tcp", devnet_config.bitcoin_node_rpc_port),
+            Some(vec![PortBinding {
+                host_ip: Some(String::from("0.0.0.0")),
+                host_port: Some(format!("{}/tcp", devnet_config.bitcoin_node_rpc_port)),
+            }]),
+        );
+
+        let bitcoind_conf = format!(
+            r#"
+server=1
+regtest=1
+rpcallowip=0.0.0.0/0
+rpcallowip=::/0
+rpcuser={}
+rpcpassword={}
+txindex=1
+listen=1
+reindex=0
+discover=0
+dns=0
+dnsseed=0
+listenonion=0
+rpcserialversion=0
+rpcworkqueue=100
+disablewallet=0
+fallbackfee=0.00001
+
+[regtest]
+bind=0.0.0.0:{}
+rpcbind=0.0.0.0:{}
+rpcport={}
+"#,
```

Comment:
Future idea: Would be cool if `devnet` could do two things WRT container configs:

1. Optionally read/ingest configs on the local filesystem for any of the containers it runs. Otherwise use the default configs like this.
2. Have a subcommand under `devnet` like `clarinet devnet write-configs` to write the default container configs to the local filesystem for better debugging visibility, and to give devs an easy starting point to modify their devnet configs for a more custom deployment.

Reply:
this configs are being written in the working dir:

```
➜  stacks-devnet-1628076160 tree .
.
├── conf
│   ├── Config.toml
│   ├── bitcoin.conf
│   └── puppet-chain.toml
├── data
│   └── 1
│       ├── bitcoin
│       │   └── regtest
|       ...
│       └── stacks
│           └── krypton
|       ....
└── devnet.log
```